### PR TITLE
update testing harness / move contact-list function to util

### PR DESCRIPTION
**Pull request by BuckarooBanzay**
Changes:
* Adds `5.7.0` to testing matrix (replaces `5.6.1` minor version)
* Moves `mail.compile_contact_list(name, selected, playernames)` to util module

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     timeout-minutes: 10
     strategy:
       matrix:
-        ENGINE_VERSION: [5.0.0, 5.1.0, 5.2.0, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.6.1, latest]
```
Comment (Athozus) [inline .github/workflows/test.yml:12]: `latest` is already there for `5.7.0`. For me, it's not necessary to replace `5.6.1` which permit another test for that release.

File: docker-compose.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ services:
   build:
    context: ./test
    args:
-    ENGINE_VERSION: ${ENGINE_VERSION:-5.5.0}
```
Comment (Athozus) [inline docker-compose.yml:8]: Good to update that.

File: util/contact.lua [hunk 3]
```diff
@@ -0,0 +1,73 @@
+
+function mail.compile_contact_list(name, selected, playernames)
```
Comment (Athozus) [inline util/contact.lua:2]: Yeah, good idea to move that func into `util/`.

File: test/Dockerfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
-ARG ENGINE_VERSION=5.5.0
```
Comment (Athozus) [inline test/Dockerfile:1]: LGTM

**Review by Athozus — COMMENTED** [review]
Good merge request just undo your `test.yml` I think. `latest` already redirects to `5.7.0`, and the idea is to check every release since `5.0.0`. Then, I will approve merge ;)

**Review by S-S-X — APPROVED** [review]
Looks good to me.

Left note about testing base releases for minor versions vs latest bug fix releases for minor versions but I guess it isn't that important, at least not until it is.

**Pull request merged**